executor/simple.go

```diff
@@ -1203,6 +1322,27 @@ func userExists(ctx sessionctx.Context, name string, host string) (bool, error)
 	return len(rows) > 0, nil
 }
 
+// use the same internal executor to read within the same transaction, otherwise same as userExists
+func userExistsInternal(sqlExecutor sqlexec.SQLExecutor, name string, host string) (bool, error) {
+	sql := new(strings.Builder)
+	sqlexec.MustFormatSQL(sql, `SELECT * FROM %n.%n WHERE User=%? AND Host=%?;`, mysql.SystemDB, mysql.UserTable, name, host)
+	recordSet, err := sqlExecutor.ExecuteInternal(context.TODO(), sql.String())
```

Comment:
Calling `ExecRestrictedStmt` can simplify this block because it doesn't need to process `recordSet`.

Reply:
`userExists()` uses `ExecRestrictedStmt` and it would not see already made changes (updates) to the system table. I assume since it using its own transaction?
That is the reason I added the userExistsInternal, to use the same `SQLExecutor`/transaction. Is that possible with the `ExecRestrictedStmt`?

Reply:
No, `ExecRestrictedStmt` uses another session. I got it.